tests/subsys/bluetooth/mesh/sensor_subsys/src/main.c

```diff
@@ -1234,6 +1234,270 @@ static void apparent_power_check(const struct bt_mesh_sensor_type *sensor_type)
 	}
 }
 
+static void electric_current_check(const struct bt_mesh_sensor_type *sensor_type)
+{
+	struct sensor_value in_value[sensor_type->channel_count];
+	uint16_t test_vector[] = {0, 1234, 5555, 11111, 65534, UINT16_MAX};
+
+	memset(in_value, 0, sizeof(in_value));
+
+	for (int i = 0; i < ARRAY_SIZE(test_vector); i++) {
+		uint16_t expected[sensor_type->channel_count];
+
+		for (int j = 0; j < sensor_type->channel_count; j++) {
+			expected[j] = test_vector[i];
+			in_value[j].val1 = test_vector[i] == UINT16_MAX ?
+					UINT16_MAX : test_vector[i] / 100;
+			in_value[j].val2 = test_vector[i] == UINT16_MAX ?
+					0 : test_vector[i] % 100 * 10000;
+		}
+
+		encoding_checking_proceed(sensor_type, in_value, expected, sizeof(expected));
+		decoding_checking_proceed(sensor_type, expected, sizeof(expected), in_value);
+	}
+}
+
+static void voltage_check(const struct bt_mesh_sensor_type *sensor_type)
+{
+	struct sensor_value in_value[sensor_type->channel_count];
+	uint16_t test_vector[] = {0, 123, 159, 160, UINT16_MAX};
+
+	memset(in_value, 0, sizeof(in_value));
+
+	for (int i = 0; i < ARRAY_SIZE(test_vector); i++) {
+		uint16_t expected[sensor_type->channel_count];
+
+		for (int j = 0; j < sensor_type->channel_count; j++) {
+			in_value[j].val1 = test_vector[i] > 159 ?
+					UINT16_MAX : test_vector[i];
+			in_value[j].val2 = 0;
+			expected[j] = test_vector[i] > 159 ?
+				UINT16_MAX : raw_scalar_value_get(test_vector[i], 1, 0, -6);
+		}
+
+		encoding_checking_proceed(sensor_type, in_value, expected, sizeof(expected));
+		decoding_checking_proceed(sensor_type, expected, sizeof(expected), in_value);
+	}
+}
+
+static void average_current_check(const struct bt_mesh_sensor_type *sensor_type)
+{
+	struct sensor_value in_value[sensor_type->channel_count];
```

Comment:
It doesn't have a lot of sense since encoding\decoding will be broken. Test vector is recalculated to packed data into a buffer that is compared to the template according to the channel number.

Reply:
I mean the template is calculated assuming that the channel number is 2.

Reply:
But if it is 1 the test will crash, but if it is 3 it will pass without being noticed.

Reply:
I guess a zassert could be added to check the count for each characteristic. This should be done in every test case, if so.

Reply:
Actually no, you will get `*** stack smashing detected ***: terminated` message and emergent test termination with error since the memory allocated for channels is used for data decoding. This checking is quite useless.

Reply:
try it in `voltage_check` function. this checking is used for two sensors with different number of channels